projects/extension/sql/incremental/003-vec-storage.sql

```diff
@@ -0,0 +1,30 @@
+
+-- switch the vector columns from storage external to storage plain
+do language plpgsql $block$
+declare
+    _sql text;
+begin
+    set local search_path = pg_catalog, pg_temp;
+    for _sql in
+    (
+        select pg_catalog.format
+        ( $sql$alter table %I.%I alter column embedding set storage plain$sql$
+        , v.target_schema
+        , v.target_table
+        )
+        from ai.vectorizer v
+        inner join pg_catalog.pg_class k on (k.relname = v.target_table)
+        inner join pg_catalog.pg_namespace n on (k.relnamespace = n.oid and n.nspname = v.target_schema)
+        inner join pg_catalog.pg_attribute a on (k.oid = a.attrelid)
+        where a.attname = 'embedding'
+        and a.attstorage != 'p' -- not plain
+        and a.atttypmod <= 2000
```

Comment:
this migration only happens if the vector dimensions are less than 2001 and storage isn't already plain